jcl/src/jdk.management/share/classes/openj9/lang/management/OpenJ9DiagnosticsMXBean.java

```diff
@@ -0,0 +1,168 @@
+/*[ENDIF]*/
+
+/**
+ * <p>
+ * This interface provides APIs to dynamically trigger Dump agents. APIs are also available to
+ * configure dump options.
+ * This MXBean reuses the methods in com.ibm.jvm.Dump API 
+ * <br>
+ * <table border="1">
+ * <caption><b>Usage example for the {@link OpenJ9DiagnosticsMXBean}</b></caption>
+ * <tr> <td>
+ * <pre>
+ * {@code
+ *   ...
+ *   try {
+ *      mxbeanName = new ObjectName("openj9.lang.management:type=OpenJ9Diagnostics");
+ *   } catch (MalformedObjectNameException e) {
+ *      // Exception Handling
+ *   }
+ *   try {
+ *      MBeanServer mbeanServer = ManagementFactory.getPlatformMBeanServer();
+ *      if (false == mbeanServer.isRegistered(mxbeanName)) {
+ *         // OpenJ9DiagnosticsMXBean not registered
+ *      }
+ *      OpenJ9DiagnosticsMXBean diagBean = JMX.newMXBeanProxy(mbeanServer, mxbeanName, OpenJ9DiagnosticsMXBean.class);
+ *   } catch (Exception e) {
+ *      // Exception Handling
+ *   }
+ * }
+ * </pre></td></tr>
+ * </table>
+ */
+public interface OpenJ9DiagnosticsMXBean extends PlatformManagedObject {
+	/**
+	 * Reset the JVM dump options to the settings specified when the JVM was started removing any additional
+	 * configuration done since then. This method may throw a DumpConfigurationUnavailableException if the dump 
+	 * configuration cannot be altered. If this occurs it will usually be because a dump event is currently being 
+	 * handled.
+	 *
+	 * @throws DumpConfigurationUnavailableException if the configuration cannot be changed because a dump is already in progress
+	 * @throws SecurityException if there is a security manager and it doesn't allow the checks required to change the dump settings
+	 */
+/*[IF Sidecar19-SE]*/
+	public void resetDumpOptions();
+/*[ELSE]
+	public void resetDumpOptions() throws DumpConfigurationUnavailableException;
+/*[ENDIF]*/
+
+	/**
+	 * This function sets options for the dump subsystem.
+	 * The dump option is passed in as a String. Use the same syntax as the -Xdump command-line option, with the 
+	 * initial -Xdump: omitted. See Using the -Xdump option as described in the section on dump agents in the 
+	 * documentation for the OpenJ9 JVM. This method may throw a DumpConfigurationUnavailableException if the dump
+	 * configuration cannot be altered.
+	 *
+	 * @param dumpOptions the options string to be set
+	 * @throws InvalidDumpOptionException if the specified dumpOptions cannot be set or is incorrect
+	 * @throws DumpConfigurationUnavailableException if the configuration cannot be changed because a dump is already in progress
+	 * @throws SecurityException if there is a security manager and it doesn't allow the checks required to change the dump settings
+	 * @throws NullPointerException if dumpOptions is null
+	 */
+/*[IF Sidecar19-SE]*/
+	public void setDumpOptions(String dumpOptions);
+/*[ELSE]
+	public void setDumpOptions(String dumpOptions) throws InvalidDumpOptionException, DumpConfigurationUnavailableException;
+/*[ENDIF]*/
```

Comment:
This pattern of declaring checked exceptions in Java 8 and no checked exceptions in Java 9 creates a source incompatibility: code that compiles with Java 8 won't compile with Java 9 (e.g. the compiler will complain about trying to catch `InvalidDumpOptionException` that isn't thrown). To create a consistent API, those exceptions from the implementation may need to be remapped to new exception classes.

Reply:
Agreed